src/casp/services/api/services/cell_operations_service.py

```diff
@@ -58,6 +59,52 @@ def authorize_model_for_user(self, user: models.User, model_name: str) -> models
             )
         return model
 
+    def __get_cell_count_from_anndata(self, file: t.BinaryIO) -> int:
+        """
+        Get the number of cells in the anndata file.
+
+        :param file: Anndata file to get the number of cells from.
+
+        :return: Number of cells in the anndata file.
+        """
+        adata = anndata.read_h5ad(file)
```

Comment:
Oh neat, that's convenient.  I'll update that

Reply:
Reading in that mode doesn't seem to work with `BinaryIO`.  I assume we have to use `BinaryIO` because a fastapi `UploadFile` isn't the same thing as a python file object, right?

Reply:
Looks like so, that's sad.:(